gerador-piadas/src/main/java/br/com/codenation/PiadaServiceImpl.java

```diff
@@ -0,0 +1,32 @@
+package br.com.codenation;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class PiadaServiceImpl implements PiadasService{
```

Comment:
opa, essa é uma conversa.
Primeiro: Por que vc tem uma interface para apenas uma única implementação?
YAGNI: You aren't gonna need it
* https://softwareengineering.stackexchange.com/questions/159813/do-i-need-to-use-an-interface-when-only-one-class-will-ever-implement-it/159821#159821
* https://softwareengineering.stackexchange.com/questions/159813/do-i-need-to-use-an-interface-when-only-one-class-will-ever-implement-it

Reply:
Eai Otavio, muito obrigado pelos comentários !!

Só uma dúvida, é bem comum encontrar exemplos de interface para apenas uma implementação quando procuramos por "Spring boot service" :

-  [](https://medium.com/@gustavo.ponce.ch/spring-boot-restful-tutorial-7f0e314242ef)

-  [](https://www.tutorialspoint.com/spring_boot/spring_boot_service_components.htm) 

Eu já vi algumas coisas relacionando isso com a injeção de dependências do spring, que você da um @Autowired na interface (igual eu acabei fazendo).  Minha dúvida é, mesmo com injeção de dependência eu não devo criar uma interface que vai ter somente uma implementação ?

Reply:
Olha, esse modo uma interface uma implementação foi algo que era obrigatório lá na época do EjB 2.0. Por questões de proxy e corba. Tecnicamente, vc não precisa de uma interface para apenas uma implementação.